program.py

```diff
@@ -0,0 +1,138 @@
+class Label:
+    """
+    Label represents a label for an instruction that
+    can be jumped to during exectuion
+    """
+
+    def __init__(self, name: str):
+        self.name = name
+
+
+class Variable:
+    """
+    Variable represents either a raw value
+    or a variable stored in memory
+    """
+
+    def __init__(self, name: str, index: int):
+        self.name = name
+        self.index = index
+
+
+class Jump:
+    """
+    Jump represents a an instruction to jump to
+    """
+
+    def __init__(self, label: Label, var: Variable, condition: int):
+        self.label = label
+        self.var = var
+        self.condition = condition
+
+
+class Operation:
+    """
+    Operation represents some operation on a variable
+    """
+
+    ASSIGNMENT = ["="]
+
+    """
+    If operator is None, then the operation is just a value (lhs)
+    """
+
+    # rhs is type Operation
+    def __init__(self, lhs: Variable, operator: str, rhs):
+
+        if operator != None:
+            # Allow for other operators to be implemented
+            if operator != Operation.ASSIGNMENT:
+                raise TypeError
+
+        self.lhs = var
+        self.operator = operator
+        self.rhs = rhs
```

Comment:
I see you made a point of saying this will probably not be used, so feel free to ignore but fyi recall that in quil, the arithmetic is implemented like 
```
ADD <variable> <variable> 
```
And assignment is 
```
MOV <source> <destination> 
```
For [example](https://github.com/not-classical/test_quil/blob/master/factoral.quil#L9). Which is to say, everything is in prefix notation (not infix notation, i.e. "left hand side/right hand side" type of parsing)

Reply:
Yeah, I just removed it for simplicity